envoy/api/v2/cds.proto

```diff
@@ -437,3 +441,8 @@ message UpstreamBindConfig {
   // The address Envoy should bind to when establishing upstream connections.
   core.Address source_address = 1;
 }
+
+message UpstreamConnectionOptions {
```

Comment:
LGTM, but I'd rename the message to just `ConnectionOptions`, since we might want to reuse this elsewhere, e.g. health check connections etc.

Reply:
I think the health checks will inherit these options from the cluster already in my implementation.
I didn't want to call it ConnectionOptions, as I suspect we'll want a similar, but not identical, message for accept()ed connections.
Perhaps ClusterConnectionOptions?